### ISSUE TITLE
Support returning multiple validation errors

### ISSUE THREAD

**Issue by fenollp** [linka-cloud/prost-validate#6]
Today, validation ends on the first validation error encountered.

This asks that all validation errors be aggregated in some kind of collection, so all validation errors (or at least, as many as possible or that makes sense) are returned.

This can be behind a generation flag so both behaviors can be generated. Generating both at once (so both ways are available at runtime) should probably not be a goal.

**Comment by Adphi** [comment]
Maybe we could add a second trait `AllValidator` or something with a `validate_all` method.

**Comment by Adphi** [comment]
@fenollp It shouldn't be too hard to implement. Do you want to give it a go?

**Comment by fenollp** [comment]
Hey! I'm down to have a go at this (and also the eventual other issues I open) but I need a hand.

Am I getting this right in that I should be adding a `validate_all` method to the trait, then edit this part for codegen https://github.com/linka-cloud/prost-validate/blob/49b5be36df98d4af5ff3b8c6a0fed0b509c53ba1/prost-validate-derive-core/src/derive.rs#L56-L70

Then I should add a new variant say `ManyErrors(Vec<Box<Error>>) to https://docs.rs/prost-validate/latest/prost_validate/errors/enum.Error.html (or should it be un-Box-ed?, or a new error type altogether?)

Lastly, should this be under a plugin option and if so, where is this handled? I see many options mixed in and am getting lost.

Thanks!

**Comment by Adphi** [comment]
Yes, but I'm not sure if `validate_all` should be an additional method or an additional trait.

Maybe an `Errors` type that is a vector of [Error](https://github.com/linka-cloud/prost-validate/blob/main/prost-validate/src/error.rs).

Defaults to generate the `validate_all` method / trait for now. We can add an option for that later if needed.

Would you be interested in setting up a short call to go over it together ?
